josh-core/src/filter/mod.rs

```diff
@@ -2097,4 +2186,90 @@ mod tests {
         // The spec should contain the chain representation
         assert!(!spec_str.is_empty());
     }
+
+    #[test]
+    fn meta_filter_parsing_test() {
```

Comment:
these tests look rather messy with all the string escaping going on; consider raw strings.